### PR TITLE
docs: Document architecture of LAPIS and surroundings

### PR DESCRIPTION
**Pull request by fengelniederhammer**
This is a draft to roughly sketch the dependencies between various data sources and artifacts. Feedback is welcome, this is supposed to be extended.

### CODE REVIEW

File: docs/developer/architectural_overview.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Architectural Overview
+
+![architecture_diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/GenSpectrum/LAPIS/documentArchitecture/docs/developer/LAPIS.puml)
```
Comment (fengelniederhammer) [inline docs/developer/architectural_overview.md:3]: - [ ] Change the URL before merge

File: docs/developer/LAPIS.puml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+package LAPIS {
+    [In Memory DB]
+    [API]
+  [Update Pipeline]
+}
+
+package "Cov-Spectrum-Server" {
+    [Server API]
+    database "???"
```
Comment (chaoran-chen) [inline docs/developer/LAPIS.puml:17]: The server provides COVID-19 case data from Our World in Data (OWID) and Pango lineage alias data. They are imported by these scripts:

- https://github.com/GenSpectrum/cov-spectrum-server/tree/develop/ext_models/ingest-owidCases
- https://github.com/GenSpectrum/cov-spectrum-server/tree/develop/ext_models/ingest-pangoLineageAliases

They are directly running on GitHub:

- https://github.com/GenSpectrum/cov-spectrum-server/actions/workflows/run_ingest-owidCases.yml
- https://github.com/GenSpectrum/cov-spectrum-server/actions/workflows/run_ingest-pangoLineageAliases.yml

File: docs/developer/LAPIS.puml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+[Server API] -down-> [Website] : additional data
+"???" -left-> [Server API] : where does the data come from?
+
+API -down-> [Website] : sequence data
+API <- [In Memory DB]
+"Postgres DB" -down-> API
+
+externalDB -down-> [Update Pipeline] : preprocessing
+[Update Pipeline] -up-> "Postgres DB" : scheduled job
+"Postgres DB" -right-> [In Memory DB] : at program start
```
Comment (chaoran-chen) [inline docs/developer/LAPIS.puml:31]:
```suggestion
"Postgres DB" -right-> [In Memory DB] : at program start / new data available
```

**Review by chaoran-chen — COMMENTED** [review]
See below. Otherwise, the diagram looks good!

**Review by chaoran-chen — APPROVED** [review]
Looks good for the start!

**Pull request merged**